dwitter/templates/base.html

```diff
@@ -36,43 +36,50 @@
   </head>
   <body class="{% block body_class %}{% endblock %}">
     <script type="text/javascript">
-      // Check for compatibility
-      if (window.localStorage) {
-        var switchTheme = function() {
-          // Shift themes
-          var theme = localStorage.getItem('theme');
-          if (!theme) {
-            localStorage.setItem('theme', 'dark');
-          } else if (theme === 'dark') {
-            localStorage.setItem('theme', 'green');
-          } else if (theme === 'green') {
-            localStorage.setItem('theme', 'hcontrast');
-          } else {
-            localStorage.setItem('theme', '');
-          }
-          location.reload();
-          // Call this function in the li to change the mode
+      (function() {
+        var themes = ['', 'dark', 'green', 'hcontrast'];
+        var themeUrls = {
+          dark: '{% static "theme_dark.css" %}',
+          green: '{% static "theme_green.css" %}',
+          hcontrast: '{% static "theme_hcontrast.css" %}'
         };
+        // Use saved theme if localStorage is available
+        var theme = window.localStorage ? localStorage.getItem('theme') : null;
+        // Ignore unknown theme
+        if (themes.indexOf(theme) == -1) {
+          theme = null;
+        }
+        // Use dark mode preference if no theme is set
+        if (typeof theme != 'string' && window.matchMedia) {
```

Comment:
Anyone who has looked at the other themes and gone back to the default will have `''` stored and won't be able to get the new automatic behavior. It might be worth being more explicit and having a 'light', 'dark' and 'automatic' theme and move anyone with `''` over to automatic? 

Could be done in a follow-up pull request

Reply:
As a one-time fix we could just use another key in local storage and ignore the old preference if it is `''` and that could easily be done in this PR. I wouldn’t add the _auto_ theme as long as it’s not displayed to the user because they’d cycle through the themes and see the “same” theme twice and wouldn’t know the difference.
 
As for long term behavior I’d prefer to move this to a place with a bit more room to show theme names and also explain what _automatic_ means.

Reply:
Yeah I agree that `''` can mean automatic, without creating a new `'automatic'` key.

But we do need a new `'light'` key.

Reply:
If we don't want the user to see a duplicate theme, then it sounds like we must either:

- Not allow them to travel to `''` (automatic), or
- Not allow them to travel to `'light'` or `'dark'`, whichever is selected by automatic, or
- Provide some additional way to toggle the automatic theme selection, so if they are on `'light'` they can get back to `''`

Reply:
I’ll get rid of `''` for now. There won’t be an *automatic* theme right now, just a "user hasn’t selected a theme" which the users with `''` will fall back to. If they explicitly choose a theme, they won’t be able to go back for now but they’ll also know how to change it manually.
Yes, there is some more UI needed to do it properly and I’m not sure it should be fit into the dropdown menu.